depends/funcs.mk

```diff
@@ -179,6 +179,8 @@ $(1)_cmake=env CC="$$($(1)_cc)" \
                CXX="$$($(1)_cxx)" \
                CXXFLAGS="$$($(1)_cppflags) $$($(1)_cxxflags)" \
                LDFLAGS="$$($(1)_ldflags)" \
+               AR="$$($(1)_ar)" \
+               RANLIB="$$($(1)_ranlib)" \
```

Comment:
In contrary to the `CC` and `CXX` environment variables, CMake [docs](https://cmake.org/cmake/help/latest/manual/cmake-env-variables.7.html) do not mention that `AR` and `RANLIB` are honored as well.

I suggest to use [`CMAKE_AR`](https://cmake.org/cmake/help/latest/variable/CMAKE_AR.html) and [`CMAKE_RANLIB`](https://cmake.org/cmake/help/latest/variable/CMAKE_RANLIB.html) instead.

Reply:
I tried both of those before landing on this change, and they didn't seem to work properly. Happy to have a look if you can show a patch that works. There's also no CMAKE_NM, so that isn't going to work if we want to pass that through as well.

Reply:
Hmm, looking at the docs and playing around, `CMAKE_AR` seems more correct to me too. It seems like it should work correctly, what was the problem?

There's also `CMAKE_C_COMPILER_AR` which may be more correct?

Reply:
There's also no CMAKE_NM, so that isn't going to work if we want to pass that through as well.

I see `CMAKE_NM` in my `CMakeCache.txt`, maybe it's just not documented for some reason?

Reply:
> Hmm, looking at the docs and playing around, `CMAKE_AR` seems more correct to me too. It seems like it should work correctly, what was the problem?

It seems `CMAKE_AR` is interpreted as relative to the build dir for some reason. What works for me is: 
```
-DCMAKE_AR=`which ar`
```

Reply:
> What works for me is:
> Bug here: https://gitlab.kitware.com/cmake/cmake/-/issues/18087

Given the bug report is 5 years old, I can't imagine CMake is going to either "fix" or change things now. I can switch to `CMAKE_` using `which` if that is what's preferred. Will add `CMAKE_NM` even if that's missing from the CMake docs. Maybe we can upstream a fix for that as well. Whatever will move things forward.

Reply:
Thanks, I think this is the way to go.